### PR TITLE
Sample generate files improvements

### PR DESCRIPTION
**Pull request by antgonza**
~https://github.com/biocore/qiita/pull/2743 should be reviewed/merged first~

### CODE REVIEW

File: qiita_db/metadata_template/base_metadata_template.py [hunk 1]
```diff
@@ -1315,7 +1331,7 @@ def _update(self, md_template):
                     "There are no differences between the data stored in the "
                     "DB and the new data provided",
                     qdb.exceptions.QiitaDBWarning)
-                return
+                return set([]), set([])
```
Comment (charles-cowart) [inline qiita_db/metadata_template/base_metadata_template.py]: Is it better to return two empty sets vs None, or (None, None), throw an Exception, etc?

File: qiita_db/metadata_template/base_metadata_template.py [hunk 2]
```diff
@@ -1396,9 +1416,9 @@ def update(self, md_template):
             # Clean and validate the metadata template given
             new_map = self._clean_validate_template(
                 md_template, self.study_id, current_columns=self.categories())
-            self._update(new_map)
+            samples, columns = self._update(new_map)
             self.validate(self.columns_restrictions)
```
Comment (charles-cowart) [inline qiita_db/metadata_template/base_metadata_template.py]: Feels like it'd be more readable to test for not None or similar here, instead of having to go down into validate(), but I defer.
